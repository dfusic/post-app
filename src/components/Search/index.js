import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';

const Search = (props) => {
  const { onChange, value, onSubmit, resetSearch } = props;

  return (
    <StyledSearch>
      <StyledSearchInput
        placeholder="Search posts from user."
        value={value}
        onChange={(e) => onChange(e)}
      />
      <StyledSearchButton onClick={onSubmit}>Search</StyledSearchButton>
      {value.length > 0 && <StyledSearchButton onClick={resetSearch}>Reset</StyledSearchButton>}
    </StyledSearch>
  );
};

const StyledSearch = styled.div`
  width: 100%;
  display: flex;
  flex-direction: row;
  width: 100%;
`;

const StyledSearchInput = styled.input`
  width: 100%;
  border: 1px solid grey;
  padding: 8px;
  font-family: 'Roboto', sans-serif;
`;
const StyledSearchButton = styled.button`
  border: 1px solid grey;
  padding: 8px;
  font-family: 'Roboto', sans-serif;
  width: 15%;
  cursor: pointer;
`;

Search.propTypes = {
  value: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
  onSubmit: PropTypes.func.isRequired,
  resetSearch: PropTypes.func.isRequired,
};

Search.displayName = 'Search';

export default Search;
