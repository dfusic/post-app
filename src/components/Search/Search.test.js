/* eslint-disable no-undef */
import React from 'react';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import Search from './index';

const SearchComponentWithoutReset = (
  <Search onChange={() => {}} value="" onSubmit={() => {}} resetSearch={() => {}} />
);

const SearchComponentWithReset = (
  <Search onChange={() => {}} value="search" onSubmit={() => {}} resetSearch={() => {}} />
);

test('renders component with and without reset', () => {
  render(SearchComponentWithoutReset);
  render(SearchComponentWithReset);
});

test('renders reset', () => {
  render(SearchComponentWithReset);
  expect(screen.getByText('Reset'));
});

test('can type into input', () => {
  render(SearchComponentWithoutReset);
  const input = screen.getByPlaceholderText('Search posts from user.');
  userEvent.type(input, 'searching...');
});

test('matches snapshot - without reset', () => {
  expect(() => {
    render(SearchComponentWithoutReset);
  }).toMatchSnapshot();
});

test('matches snapshot - with reset', () => {
  expect(() => {
    render(SearchComponentWithReset);
  }).toMatchSnapshot();
});
