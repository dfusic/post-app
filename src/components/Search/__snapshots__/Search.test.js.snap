// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot - with reset 1`] = `[Function]`;

exports[`matches snapshot - without reset 1`] = `[Function]`;
